### ISSUE TITLE
[Choice of eta used in Lemma 2 (eq.11)]

### ISSUE THREAD

**Issue by tingwl0122** [WANGBohaO-jpg/MSL#4]
Hi Authors,
  Thanks for sharing the code for the paper. This is a really good work that inspires my research a lot.

  I have one question about the choice of the eta (which is set to 0.25) in the exp setting.
  From the adaptive-tau paper (their Lemma 2), shouldn't it be set as 0.5 to reach the largest upper bound of expected gradient norm?
  Or is there any specific reason/derivation that make you choose 0.25 as the default value?

  It will be great to learn more about this design choice.  Thank you in advance!

**Comment by WANGBohaO-jpg** [comment]
Thank you very much for your interest in our work and your thoughtful question!

You are correct that in the adaptive-tau paper, Lemma 2 suggests setting eta to 0.5 in order to maximize the expected gradient. However, there is a key difference between our work and the adaptive-tau paper regarding the choice of loss function.

Specifically, the adaptive-tau paper uses the standard softmax loss (see Eq. 1 in their paper), while our work adopts the log-softmax loss (see Eq. 1 in our paper). Due to this difference, the maximum gradient value with respect to eta is not the same for the two types of loss functions. In the adaptive-tau paper, the maximum is achieved when eta = 0.5, but for the log-softmax loss used in our work, the maximum gradient actually occurs at eta = 0. The choice of eta = 0.25 in our experiments is based on empirical considerations, balancing stability and performance.

**Comment by tingwl0122** [comment]
Thanks for the clarification. Did you ever tried eta = 0, where the gradient norm can have largest upper bound? It will be interesting to investigate the empirical performance under different etas.

Since the paper is already accepted (and I guess camera-ready version is also submitted), I will suggest the author provide a bit more information on this in the github README or anywhere else. Otherwise it feels like an additional hidden hyper-parameter that contradicts to the paper's "automatically computed temperature" claim. And this 0.25 might not be a good default value for other model/dataset settings, which requires certain amount of tuning.

Thanks!
